libraries/AP_HAL/AnalogIn.h

```diff
@@ -42,6 +42,16 @@ class AP_HAL::AnalogIn {
     // failures can still be diagnosed
     virtual uint16_t accumulated_power_status_flags(void) const { return 0; }
 
+    // this enum class is 1:1 with MAVLink's MAV_POWER_STATUS enumeration!
+    enum class PowerStatusFlag : uint16_t {
+        BRICK_VALID = 1,                  // main brick power supply valid
+        SERVO_VALID = 2,                  // main servo power supply valid for FMU
+        USB_CONNECTED = 4,                // USB power is connected
+        PERIPH_OVERCURRENT = 8,           // peripheral supply is in over-current state
+        PERIPH_HIPOWER_OVERCURRENT = 16,  // hi-power peripheral supply is in over-current state
+        CHANGED = 32,                     // Power status has changed since boot
+    };
+
```

Comment:
Can you add a static assert to make sure they match?

Reply:
> Can you add a static assert to make sure they match?

I can (and I did for an GPS patch), but I don't want to :-)

These are so static - and only used for documentation - so I got lazy.  As soon as we used these in the actual ArduPilot code they'd definitely need that assertion.

Also - PRs welcome ;-)